datasource/http/data.go

```diff
@@ -62,6 +66,25 @@ func (d *Datasource) Configure(raws ...interface{}) error {
 			fmt.Errorf("the `url` must be specified"))
 	}
 
+	// Default to GET if no method is specified
+	if d.config.Method == "" {
+		d.config.Method = "GET"
+	}
+
+	// Check if the input is in the list of allowed methods
+	validMethod := false
+	allowedMethods := []string{"HEAD", "GET", "POST", "PUT"}
```

Comment:
I limited it because it is limited in terraform (there even `PUT` is not allowed. I'm fully behind being more open, but there still should be some validation. Should we allow all HTTP methods, including `CONNECT` and `TRACE` from the [standard](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods)?  I don't know if these have to be handled diffferently.